### PR TITLE
Stamina projectiles no longer set off grenades and flamethrowers

### LINKED ISSUE #28481: the disabler beam hits Unknown's X4, setting it off! What a shot!
Blowing up grenades in people's hands should check for nodamage, similar to welding fuel tank explosions.

Also, the first letter should be capitalized!

This is probably a shitty feature to begin with, it's literally a prob(15) to just explode and die every time someone shoots you if you're holding any form of grenade.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #28481

@KorPhaeron because I'm not sure I agree with this change

### CODE REVIEW

**Comment by kevinz000** [comment]
wtf fun removal

**Comment by Repukan** [comment]
Hugbox

**Comment by Robustin** [comment]
Common sense change threatened to be derailed by memesters.

**Comment by octareenroon91** [comment]
It's a matter of consistency: these are the same rules as welding-fuel tanks.

**Comment by bgobandit** [comment]
how is this not a balance change

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -353,7 +353,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 // afterattack() and attack() prototypes moved to _onclick/item_attack.dm for consistency
 
-/obj/item/proc/hit_reaction(mob/living/carbon/human/owner, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+/obj/item/proc/hit_reaction(mob/living/carbon/human/owner, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK, atom/movable/AM)
```
Comment (RemieRichards) [inline code/game/objects/items.dm]: I prefer @kevinz000 attempt.
put the new arg in second place, and update all calls.

If hit reaction had been perfect from the start, that would be the order.
Also, don't call it ```AM``` give it a descriptive name (like, ```reaction_source```)

**Comment by Cheridan** [comment]
I agree with @octareenroon91, this is more of a consistency fix than anything (and this has probably happened like once ever anyway so w/e)

I do like lumi's idea in the issue report about making it based off where you're aiming instead of just prob(15) as well

File: code/game/objects/items.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/objects/items.dm]: also you can't have a non named argument after named arguments like this

**Comment by optimumtact** [comment]
my pr doesn't actually fix this, but it's a better implementation of the rename that takes into account that not all check shield calls pass in an atom when they should

you should rebase yours on top of mine

**Comment by Cyberboss** [comment]
The def_zone thing isn't feasible because of how check_shields works (called by many things which don't have targets e.g. throwing)

**Comment by optimumtact** [comment]
Yeah honestly it's all a bit messy, the whole attack path could do with a refactor to ensure all the variables are passed all the way through

**Pull request merged**